pdr_backend/lake/gql_data_factory.py

```diff
@@ -50,37 +55,32 @@ def __init__(self, ppss: PPSS):
 
         # configure all tables that will be recorded onto lake
         self.record_config = {
-            "pdr_predictions": {
-                "fetch_fn": get_pdr_predictions_df,
-                "schema": predictions_schema,
-                "config": {
-                    "contract_list": contract_list,
-                },
-            },
-            "pdr_subscriptions": {
-                "fetch_fn": get_pdr_subscriptions_df,
-                "schema": subscriptions_schema,
-                "config": {
-                    "contract_list": contract_list,
-                },
-            },
-            "pdr_truevals": {
-                "fetch_fn": get_pdr_truevals_df,
-                "schema": truevals_schema,
-                "config": {
-                    "contract_list": contract_list,
-                },
+            "tables": {
+                "pdr_predictions": Table(
+                    predictions_table_name,
+                    predictions_schema,
+                    get_pdr_predictions_df,
+                    ppss,
+                ),
+                "pdr_subscriptions": Table(
+                    subscriptions_table_name,
+                    subscriptions_schema,
+                    get_pdr_subscriptions_df,
```

Comment:
Please remove update() function from Table().

Make that a responsibility of GQLDataFactory, OHLCVDataFactory, or ETL.
Rather, have Table be more of an abstract class.

Think of a Table in a DB... It's dumb and abstract. It's only responsibility is to provide an interface for table functions. Schema, CRUD, querying-type functionality.